### ISSUE TITLE
Usernames

### ISSUE THREAD

**Issue by Thaodan** [ars3niy/tdlib-purple#68]
Does this plugin support the use of telegram usernames instead of display names compared to telegram-purple?

**Comment by ars3niy** [comment]
No, the only way to see the username right now is by retrieving user information (for example right click, "Get Info" in pidgin).

One can consider an account option to display usernames instead of first and last names as buddy aliases (falling back to first and last name if there is no username). However, it gets tricky if a custom alias is assigned to a buddy in contact list – I guess one could keep track of which buddies have custom alias and which don't using purple_blist_node_set_string, but it's a lot of work and I'm lazy.

**Comment by Thaodan** [comment]
Could you use a layer a approach where the alias takes preference over the username and the username takes preference over first-lastname?
The thing is without displaynames group chats much harder to communicate because mentions are broken.

Anyway I understand but thanks for your work.

**Comment by ars3niy** [comment]
How do mentions work? Do you just write @username in plain text of a message and the server will figure it out?

Incidentally, the plugin won't do anything in particular if you get mentioned...

**Comment by Thaodan** [comment]
Yes that exactly how they work. The plugin doesn't need to do anything except maybe send a notification.
The issue is that the auto completion for users in chat and the group chat member list is the list where the user gets all chat participants. These should contain only the usernames to allow mentions to work. I don't know how exactly purple works with usernames but full name should be in something like user details next to the name.

**Comment by ars3niy** [comment]
I guess one can have displayed names for users with usernames be not just equal to usernames, but also contain leading @ sign. If that what you have in mind? Then in pidgin for example, you would type `@use` and press tab, and it will get completed to `@username: ` which qualifies as a mention.

**Comment by Thaodan** [comment]
Yes but with the actual telegram usernames. Currently in this and the old plugin the tab complete is useless because the mention isn't working.
Even if the @ wasn't included it would be still be better. Also some like bitlbee-discord converts Irc-style mentions like used in purple to @ style mentions used in discord. A similar feature could be useful here.
